lib/argument_builder.coffee

```diff
@@ -0,0 +1,48 @@
+_ = require 'lodash'
+fs = require 'fs'
+minimist = require 'minimist'
+path = require 'path'
+TestFinder = require './test_finder'
+
+
+class ArgumentBuilder
+
+  constructor: ({@argv, @cwd}) ->
+
+
+  build: (done) ->
+    @_loadConfig (err, config) =>
+      if err then return done err
+      @_getTestFiles config, (err, files) =>
+        if err then return done err
+        done null, @_buildDefault(config).concat(@argv).concat(files)
+
+
+  _buildDefault: ({mochaOptions, testHelpers}) ->
+    args = []
+    for own key, value of mochaOptions
+      args.push "--#{key}"
```

Comment:
In general, Node land has are a ton of libraries out there to do things like what you do here. Maybe you can use https://www.npmjs.com/package/dargs here?